code/modules/vending/clothingbooth.dm

```diff
@@ -78,11 +80,22 @@ var/list/clothingbooth_items = list()
 		if(!(usr in src.contents))
 			return
 		var/itempath = text2path(href_list["path"])
-		switch(href_list["command"])
+		var/command = href_list["command"]
+
+		if(cb_item.colored)
+			var/newColor = input("Pick a color:","Clothing color", src.selected_color) as null|color
+			if(newColor != src.selected_color)
+				command="render"
+			src.selected_color = newColor
+
+		switch(command)
 			if("spawn")
 				if(text2num_safe(cb_item.cost) <= src.money)
 					money -= text2num_safe(cb_item.cost)
-					usr.put_in_hand_or_drop(new itempath(src))
+					var/obj/item/vended_item = new itempath(src)
+					if(src.selected_color)
+						vended_item.color = src.selected_color
```

Comment:
Yeah like this shouldn't just be flat color overlays because it looks bad, it should either be hand-done or programmatically done with actual mapping of colors to another using the hsv colormapping procs